src/arkode/arkode_lsrkstep.c

```diff
@@ -1593,98 +2205,98 @@ sunbooleantype lsrkStep_CheckNVector(N_Vector tmpl)
       (tmpl->ops->nvscale == NULL) || (tmpl->ops->nvwrmsnorm == NULL) ||
       (tmpl->ops->nvspace == NULL))
   {
-    return (SUNFALSE);
+    return SUNFALSE;
   }
-  return (SUNTRUE);
+  return SUNTRUE;
 }
 
 /*---------------------------------------------------------------
-  lsrkStep_ComputeNewDomEig:
+  lsrkStep_DomEigUpdateLogic:
 
-  This routine computes new dom_eig and returns SUN_SUCCESS.
+  This routine checks if the step is accepted or not and reassigns
+  the dom_eig update flags accordingly.
   ---------------------------------------------------------------*/
 
-int lsrkStep_ComputeNewDomEig(ARKodeMem ark_mem, ARKodeLSRKStepMem step_mem)
+void lsrkStep_DomEigUpdateLogic(ARKodeMem ark_mem, ARKodeLSRKStepMem step_mem,
+                                sunrealtype dsm)
 {
-  int retval = SUN_SUCCESS;
-
-  if ((step_mem->is_ext_dom_eig))
+  if (dsm <= ONE)
   {
-    retval = step_mem->extDomEig(ark_mem->tn, ark_mem->ycur, ark_mem->fn,
-                                 &step_mem->lambdaR, &step_mem->lambdaI,
-                                 ark_mem->user_data, ark_mem->tempv1,
-                                 ark_mem->tempv2, ark_mem->tempv3);
-    step_mem->num_dom_eig_updates++;
-    if (retval != ARK_SUCCESS)
-    {
-      arkProcessError(ark_mem, ARK_DOMEIG_FAIL, __LINE__, __func__, __FILE__,
-                      "Unable to estimate the dominant eigenvalue");
-      return (ARK_DOMEIG_FAIL);
-    }
+    N_VScale(ONE, ark_mem->tempv2, ark_mem->fn);
+    ark_mem->fn_is_current = SUNTRUE;
 
-    if (step_mem->lambdaR * ark_mem->h > ZERO)
-    {
-      arkProcessError(NULL, ARK_DOMEIG_FAIL, __LINE__, __func__, __FILE__,
-                      "lambdaR*h must be nonpositive");
-      return (ARK_DOMEIG_FAIL);
-    }
-    else if (step_mem->lambdaR == 0 && SUNRabs(step_mem->lambdaI) > 0)
+    step_mem->dom_eig_is_current = (step_mem->const_Jac == SUNTRUE);
+    step_mem->dom_eig_nst = (step_mem->dom_eig_nst + 1) % step_mem->dom_eig_freq;
+    step_mem->dom_eig_update = SUNFALSE;
+    if (step_mem->dom_eig_nst == 0)
     {
-      arkProcessError(NULL, ARK_DOMEIG_FAIL, __LINE__, __func__, __FILE__,
-                      "DomEig cannot be purely imaginary");
-      return (ARK_DOMEIG_FAIL);
+      step_mem->dom_eig_update = !step_mem->dom_eig_is_current;
     }
-
-    step_mem->lambdaR *= step_mem->dom_eig_sfty;
-    step_mem->lambdaI *= step_mem->dom_eig_sfty;
-    step_mem->sprad =
-      SUNRsqrt(SUNSQR(step_mem->lambdaR) + SUNSQR(step_mem->lambdaI));
   }
   else
   {
-    arkProcessError(NULL, ARK_DOMEIG_FAIL, __LINE__, __func__, __FILE__,
-                    "Internal dom_eig is not supported yet!");
-    return (ARK_DOMEIG_FAIL);
+    step_mem->dom_eig_update = !step_mem->dom_eig_is_current;
+    step_mem->dom_eig_nst    = 0;
```

Comment:
It's sometimes helpful to know what step the last update was so rather than resetting this to `0` and using `%`, mirror the checks used for `msbj` and `msbp`